bouncer/tests/broker_level_screening.ts

```diff
@@ -56,36 +55,96 @@ async function newAssetAddress(asset: InternalAsset, seed = null): Promise<strin
+      .markTransactionAsTainted(memoryRepresentationTxId)
       .signAndSend(broker, { nonce: -1 }, handleSubstrateError(chainflip)),
   );
 }
 
 /**
- * Pauses or resumes the bitcoin block production. We send a command to the docker container to start or stop mining blocks.
+ * Ensures that the deposit-monitor is running and healthy.
  *
- * @param pause - Whether to pause or resume the block production.
- * @returns - Whether the command was successful.
  */
-function pauseBtcBlockProduction(pause: boolean): boolean {
-  try {
-    execSync(
-      pause
-        ? 'docker exec bitcoin rm /root/mine_blocks'
-        : 'docker exec bitcoin touch /root/mine_blocks',
-    );
-    return true;
-  } catch (error) {
-    console.error(error);
-    return false;
+async function ensureDepositMonitorHealth() {
+  const headers: Headers = new Headers();
+  headers.set('Content-Type', 'application/json');
+  headers.set('Accept', 'application/json');
+
+  const request: RequestInfo = new Request('http://127.0.0.1:6060/health', {
+    method: 'GET',
+    headers,
+  });
+
+  let responseBody;
+  for (let i = 0; i < 10; i++) {
+    let res;
+    try {
+      res = await fetch(request);
+    } catch {
+      testBrokerLevelScreening.log('Could not connect to deposit monitor, retrying.');
+      await sleep(1000);
+    }
+
+    if (res) {
+      const body = await res.json();
+
+      if (body.starting === false) {
+        responseBody = body;
+        break;
+      } else {
+        testBrokerLevelScreening.log('Deposit monitor is starting...');
+        await sleep(500);
+      }
+    }
+  }
+
+  if (responseBody === undefined) {
+    throw new Error('Could not ensure that deposit monitor is running.');
   }
+
+  const body = responseBody;
+  const health = body.all_processors;
+  testBrokerLevelScreening.log('Deposit monitor health: ' + health);
+  if (!health) {
+    testBrokerLevelScreening.log('Deposit monitor health response is:  ' + JSON.stringify(body));
+    throw new Error('Could not ensure that deposit monitor is healthy.');
+  }
+  return health;
+}
+
+/**
+ * Call the deposit-monitor to set risk score of given transaction in mock analysis provider.
+ * @param txid Hash of the transaction we want to report.
+ * @param score Risk score for this transaction. Can be in range [0.0, 10.0].
+ */
+function setTxRiskScore(txid: string, score: number) {
+  const headers: Headers = new Headers();
+  headers.set('Content-Type', 'application/json');
+  headers.set('Accept', 'application/json');
+  const request: RequestInfo = new Request('http://127.0.0.1:6070/riskscore', {
+    method: 'POST',
+    headers,
+    body: JSON.stringify([
+      txid,
+      {
+        risk_score: { Score: score },
+        unknown_contribution_percentage: 0.0,
+        analysis_provider: 'elliptic_analysis_provider',
+      },
+    ]),
+  });
+
+  return fetch(request).then((res) =>
+    testBrokerLevelScreening.log('got response' + JSON.stringify(res)),
+  );
```

Comment:
can probably use the `axios` library instead to remove some of the boilerplate here

Reply:
I'll look into it!

Reply:
Done, yes, it helps a bit.